docs/drafts/derive-serviceaccount.md

```diff
@@ -0,0 +1,355 @@
+In order to derive the minimal RBAC for the installer service account, you must specify the following permissions:
+* ClusterRole with all the roles specified in the bundle ClusterServiceVersion.
+* ClusterExtension finalizer
+* Allow ClusterExtenstion to set blockOwnerDeletion ownerReferences
+* create the controller deployment
+* create the ClusterResourceDefnitions
+* create the other manifest objects
+* create the necessary Cluster/Roles for the controller to be able to perform its job.
+* get, list, watch, update, patch, delete the specific resources that get created
+* update finalizers on the ClusterExtension to be able to set blockOwnerDeletion and ownerReferences
+
+
+The following ClusterRole rules are needed:
+# Allow ClusterExtension to set blockOwnerDeletion ownerReferences
+- apiGroups: [olm.operatorframework.io]
+  resources: [clusterextensions/finalizers]
+  verbs: [update]
+  resourceNames: [<cluster-extension-name>]
+# Manage CRDs
+- apiGroups: [apiextensions.k8s.io]
+  resources: [customresourcedefinitions]
+  verbs: [create, list, watch]
+- apiGroups: [apiextensions.k8s.io]
+  resources: [customresourcedefinitions]
+  verbs: [get, update, patch, delete]
+  resourceNames: [<crd name 1>, ..., <crd name n>]
+# Manage ClusterRoles
+- apiGroups: [rbac.authorization.k8s.io]
+  resources: [clusterroles]
+  verbs: [create, list, watch]
+- apiGroups: [rbac.authorization.k8s.io]
+  resources: [clusterroles]
+  verbs: [get, update, patch, delete]
+  resourceNames: [<generated cluster role 1>, ..., <generated cluster role n>, <manifest cluster role name 1>, ..., <manifest cluster role name n>]
+# Manage ClusterRoleBindings
+- apiGroups: [rbac.authorization.k8s.io]
+  resources: [clusterrolebindings]
+  verbs: [create, list, watch]
+- apiGroups: [rbac.authorization.k8s.io]
+  resources: [clusterrolebindings]
+  verbs: [get, update, patch, delete]
+  resourceNames: [<generated cluster role 1>, ..., <generated cluster role n>, <manifest cluster role binding name 1>, ..., <manifest cluster role binding name n>]
+* Rules defined in the CSV under `.spec.install.clusterPermissions` and `.spec.install.permissions`
+* Rules to manage any other cluster-scoped resources shipped with the bundle
+* Rules to manage any other namespace-scoped resources
+* Rules to manage the deployment defined in the ClusterServiceVersion
+* Rules to manage the service account used for the deployment
+- apiGroups: [apps]
+  resources: [deployments]
+  verbs: [create, list, watch]
+
+- apiGroups: [apps]
+  resources: [deployments]
+  verbs: [get, update, patch, delete]
+  resourceNames: [argocd-operator-controller-manager]
+
+- apiGroups: [""]
+  resources: [serviceaccounts]
+  verbs: [create, list, watch]
+
+- apiGroups: [""]
+  resources: [serviceaccounts]
+  verbs: [get, update, patch, delete]
+  resourceNames: [argocd-operator-controller-manager]
+```
+
+Below is an example of the argocd installer with the necessary RBAC to deploy the ArgoCD ClusterExtension:
```

Comment:
`Below is an example of the argocd installer with the necessary RBAC to deploy the ArgoCD ClusterExtension:` should be outside of the yaml block.

Reply:
fixed